packages/ui/client/composables/explorer/search.ts

```diff
@@ -0,0 +1,148 @@
+import type { Task } from '@vitest/runner'
+import type { Ref } from 'vue'
+import {
+  allExpanded,
+  failedFilter,
+  filter,
+  filteredFiles,
+  isFiltered,
+  isFilteredByStatus,
+  search,
+  skipFilter,
+  successFilter,
+} from './state'
+import type { TreeTaskFilter } from '~/composables/explorer/tree'
+import { taskTree, uiEntries } from '~/composables/explorer/tree'
+import { caseInsensitiveMatch } from '~/utils/task'
+import { client } from '~/composables/client'
+
+export function useSearch(searchBox: Ref<HTMLDivElement | undefined>) {
+  const disableFilter = computed(() => {
+    if (isFilteredByStatus.value)
+      return false
+
+    return !filter.onlyTests
+  })
+  const disableClearSearch = computed(() => search.value === '')
+
+  const filesTotal = computed<
+    { failed: number; success: number; skipped: number; running: number }
+  >(() => {
+    if (isFiltered.value) {
+      const data = filteredFiles.value.reduce((acc, task) => {
+        if (task.result?.state === 'fail')
+          acc.failed++
+        else if (task.result?.state === 'pass')
+          acc.success++
+        else if (task.mode === 'skip' || task.mode === 'todo')
+          acc.skipped++
+
+        return acc
+      }, { failed: 0, success: 0, skipped: 0, running: 0 })
+      data.running = filteredFiles.value.length - data.failed - data.success - data.skipped
+      return data
+    }
+
+    return {
+      failed: taskTree.summary.filesFailed,
+      success: taskTree.summary.filesSuccess,
+      skipped: taskTree.summary.filesSkipped,
+      running: taskTree.summary.filesRunning,
```

Comment:
review logic, should be test and no files: move the logic to the tree to collect the state and rename `filteredFiles` back to `filteredTests`